sdk/cosmosdb/cosmos/src/queryExecutionContext/EndpointComponent/GroupByEndpointComponent.ts

```diff
@@ -52,7 +52,11 @@ export class GroupByEndpointComponent implements ExecutionContext {
         if (aggregators) {
           // Iterator over all results in the payload
           Object.keys(payload).map((key) => {
-            const aggregateResult = extractAggregateResult(payload[key]);
+            // in case the value of a group is null make sure we create a dummy payload with item2==null
+            const effectiveGroupByValue = payload[key]
+              ? payload[key]
+              : new Map().set("item2", null);
+            const aggregateResult = extractAggregateResult(effectiveGroupByValue);
```

Comment:
maybe just add a null check inside the extractAggregateResult?

Reply:
no - that one is used in other places - and the fix-up logic really belongs here - the extractxxx shoudln't mutate the payload IMO